tests/e2e/files/rulebooks/operators/test_relational_operators.yml

```diff
@@ -0,0 +1,224 @@
+            square_root_of_2: 1.41
+          - id: "testcase #16"
+            square_root_of_2: 1.41421356237
+          - id: "testcase #16"
+            square_root_of_2: 1.41421356238
+
+          - id: "testcase #11"
+            myboolean: true
+          - id: "testcase #17"
+            myboolean: false
+
+          - id: "testcase #18"
+            nullified: null
+
+          - id: "testcase #19"
+            nullable: null
+
+          - id: "finish"
+            finish: true
+  rules:
+    - name: "Testcase #1"
+      condition: >
+        not event.test_bool and
+        not (event.test_str == "Goodbye" or event.test_str != "Hello there")
+      action:
+        debug:
+          msg: "Output for testcase #1"
+
+    - name: "Testcase #2"
+      condition:
+        all:
+          - not (event.test_int > 80 and event.test_int < 79)
+          - not (event.test_float <= 3.14 or event.test_float >= 4.5)
+      action:
+        debug:
+          msg: "Output for testcase #2"
+
+    - name: "Testcase #3"
+      condition: event.motto != "Fire and Blood"
+      action:
+        debug:
+          msg: "Output for testcase #3"
+
+    - name: "Testcase #4"
+      condition: 73 != event.universe_meaning
+      action:
+        debug:
+          msg: "Output for testcase #4"
+
+    - name: "Testcase #5"
+      condition: true != event.flat_earth
+      action:
+        debug:
+          msg: "Output for testcase #5"
+
+    - name: "Testcase #6"
+      condition: event.number_of_swords < 200.001
+      action:
+        debug:
+          msg: "Output for testcase #6"
+
+    - name: "Testcase #7"
+      condition: event.number_of_dragons < 4
+      action:
+        debug:
+          msg: "Output for testcase #7"
+
+    - name: "Testcase #8"
+      condition: event.numbers.nested.data <= -10
+      action:
+        debug:
+          msg: "Output for testcase #8"
+
+    - name: "Testcase #9"
+      condition: event.pi <= 3.1415
+      action:
+        debug:
+          msg: "Output for testcase #9"
+
+    - name: "Testcase #10"
+      condition: event.pi_for_engineers == 3.0
+      action:
+        debug:
+          msg: "Output for testcase #10"
+
+    - name: "Testcase #11"
+      condition: event.stark_motto == "Winter\nis\ncoming"
+      action:
+        debug:
+          msg: "Output for testcase #11"
+
+    - name: "Testcase #12"
+      condition: 3 == event.e_for_engineers
+      action:
+        debug:
+          msg: "Output for testcase #12"
+
+    - name: "Testcase #13"
+      condition: event.intelligent_design == false
+      action:
+        debug:
+          msg: "Output for testcase #13"
+
+    - name: "Testcase #14"
+      condition: event.best_number > vars.universe_meaning
+      action:
+        debug:
+          msg: "Output for testcase #14"
+
+    - name: "Testcase #15"
+      condition: event.golden_ratio > 1.6180339887
+      action:
+        debug:
+          msg: "Output for testcase #15"
+
+    - name: "Testcase #16"
```

Comment:
There are more potential combinations that are not covered, we can not cover the whole matrix, at some point we test positions, data types, variables and so on. If you have some particular concern with this specific case, I can add it, otherwise I would leave as it is now, I believe we have a reasonable coverage of use cases.

Reply:
@Alex-Izquierdo I agree that covering the entire matrix of combinations is out of scope, however this particular operator is not tested at all with the int data type which will be a common use case in customer environments. Floats on the other hand are covered in two test cases, so we have good coverage there.

Reply:
We also need test cases to validate and close [AAP-9095](https://issues.redhat.com/browse/AAP-9095).

Reply:
This is already covered here: https://github.com/ansible/ansible-rulebook/pull/381/files#diff-c4a65780a1f09e2ae6275fffc03b6fcb9f0144968c1c3e56cfb27fe5354bb99eR125 as well as ints in that rulebook.